taskvine/src/manager/vine_file.c

```diff
@@ -143,6 +143,22 @@ struct vine_file *vine_file_clone(struct vine_file *f)
 	return f;
 }
 
+/* Make a URL reference to a file source*/
+
+char *vine_file_make_file_url(const char *source)
+{
+	char *abs_path = strcat(path_getcwd(), "/");
```

Comment:
As a general rule, it's not safe to modify strings returned from functions.  I have a hunch this `strcat` is part of your memory problems, because it is attempting to append to a string allocated to a specific size.

Suggest that you just use something like `result = string_format("file:/%s/%s",abs_path,source);` to combine everything together into a new string.

And as an aside, the various `path` functions should almost certainly return `const char *` instead to discourage the caller from doing this sort of thing.